File: samples/client/echo_api/java/feign-gson/src/main/java/org/openapitools/client/api/PathApi.java
package org.openapitools.client.api;

import org.openapitools.client.ApiClient;
import org.openapitools.client.EncodingUtils;
import org.openapitools.client.model.ApiResponse;

import org.openapitools.client.model.StringEnumRef;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import feign.*;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen", comments = "Generator version: 7.14.0-SNAPSHOT")
public interface PathApi extends ApiClient.Api {


  /**
   * Test path parameter(s)
   * Test path parameter(s)
   * @param pathString  (required)
   * @param pathInteger  (required)
   * @param enumNonrefStringPath  (required)
   * @param enumRefStringPath  (required)
   * @return String
   */
  @RequestLine("GET /path/string/{pathString}/integer/{pathInteger}/{enumNonrefStringPath}/{enumRefStringPath}")
  @Headers({
    "Accept: text/plain",
  })
  String testsPathStringPathStringIntegerPathIntegerEnumNonrefStringPathEnumRefStringPath(@Param("pathString") @javax.annotation.Nonnull String pathString, @Param("pathInteger") @javax.annotation.Nonnull Integer pathInteger, @Param("enumNonrefStringPath") @javax.annotation.Nonnull String enumNonrefStringPath, @Param("enumRefStringPath") @javax.annotation.Nonnull StringEnumRef enumRefStringPath);

  /**
   * Test path parameter(s)
   * Similar to <code>testsPathStringPathStringIntegerPathIntegerEnumNonrefStringPathEnumRefStringPath</code> but it also returns the http response headers .
   * Test path parameter(s)
   * @param pathString  (required)
   * @param pathInteger  (required)
   * @param enumNonrefStringPath  (required)
   * @param enumRefStringPath  (required)
   * @return A ApiResponse that wraps the response boyd and the http headers.
   */
  @RequestLine("GET /path/string/{pathString}/integer/{pathInteger}/{enumNonrefStringPath}/{enumRefStringPath}")
  @Headers({
    "Accept: text/plain",
  })
  ApiResponse<String> testsPathStringPathStringIntegerPathIntegerEnumNonrefStringPathEnumRefStringPathWithHttpInfo(@Param("pathString") @javax.annotation.Nonnull String pathString, @Param("pathInteger") @javax.annotation.Nonnull Integer pathInteger, @Param("enumNonrefStringPath") @javax.annotation.Nonnull String enumNonrefStringPath, @Param("enumRefStringPath") @javax.annotation.Nonnull StringEnumRef enumRefStringPath);


}

File: samples/client/echo_api/kotlin-model-prefix-type-mappings/src/main/kotlin/org/openapitools/client/apis/PathApi.kt
package org.openapitools.client.apis

import org.openapitools.client.infrastructure.CollectionFormats.*
import retrofit2.http.*
import retrofit2.Response
import okhttp3.RequestBody
import com.google.gson.annotations.SerializedName

import org.openapitools.client.models.ApiStringEnumRef

interface PathApi {

    /**
    * enum for parameter enumNonrefStringPath
    */
    enum class EnumNonrefStringPathTestsPathStringPathStringIntegerPathIntegerEnumNonrefStringPathEnumRefStringPath(val value: String) {
        @SerializedName(value = "success") SUCCESS("success"),
        @SerializedName(value = "failure") FAILURE("failure"),
        @SerializedName(value = "unclassified") UNCLASSIFIED("unclassified")
    }

    /**
     * Test path parameter(s)
     * Test path parameter(s)
     * Responses:
     *  - 200: Successful operation
     *
     * @param pathString 
     * @param pathInteger 
     * @param enumNonrefStringPath 
     * @param enumRefStringPath 
     * @return [String]
     */
    @GET("path/string/{path_string}/integer/{path_integer}/{enum_nonref_string_path}/{enum_ref_string_path}")
    suspend fun testsPathStringPathStringIntegerPathIntegerEnumNonrefStringPathEnumRefStringPath(@Path("path_string") pathString: String, @Path("path_integer") pathInteger: Int, @Path("enum_nonref_string_path") enumNonrefStringPath: String, @Path("enum_ref_string_path") enumRefStringPath: ApiStringEnumRef): Response<String>

}
